### DISCUSSION TITLE
Resizing the game's window in fullscreen resizes all windows in the same display

### DISCUSSION THREAD

**Discussion by Rian8337** [ppy/osu#17825] [Q&A]
When I was testing for replications of #17380, I noticed that when resizing the game's resolution, all windows that are opened in the same display as the game gets resized as well. I'm not sure if this is an osu!-side issue or SDL's (I haven't looked at issues/discussions from osu!framework and SDL, and I'm not familiar with the inner workings of SDL).

I have tested this with stable and wasn't able to replicate the behavior.

I'll outline my test procedures below with a `1920x1080` native resolution monitor. Keep in mind that all tabbing in/out, resizing, and game opening processes were affected by the behavior described in #17380, but I decided to make a new discussion as this appears to be a separate problem in the logic behind.

<details>
<summary>Test 1</summary>

1. Opened Microsoft Edge, put to maximized size.
2. Opened the game at `Default` resolution and fullscreen initially, then changed resolution to `1680x1050`. The game's window got resized.
3. Tabbed out from the game, Microsoft Edge's window got resized to `1680x1050` as well.
4. Tabbed into the game, changed resolution to `Default`. The game's window got resized.
5. Tabbed out from the game, Microsoft Edge's window stays at `1680x1050` (I expected that this is a behavior from Windows itself, but I'm not completely sure).
6. Repeated step 4, but with changing the game's resolution to `1920x1080` instead of `Default`. Behavior in step 5 stays the same.
7. Repeated all steps, but with Microsoft Edge and Discord opened. Behavior in all steps stay the same for their respective window.
</details>

<details>
<summary>Test 2</summary>

1. Opened Microsoft Edge, put to maximized size.
2. Opened the game at `1680x1050` resolution and fullscreen initially.
3. Tabbed out from the game, Microsoft Edge's window got resized to `1680x1050`.
4. Repeated all steps, but with Microsoft Edge and Discord opened. Behavior in all steps stay the same for their respective window.
</details>

Logs:
[performance-draw.log](https://github.com/ppy/osu/files/8494278/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/8494280/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/8494281/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/8494282/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8494283/updater.log)
[database.log](https://github.com/ppy/osu/files/8494284/database.log)
[network.log](https://github.com/ppy/osu/files/8494285/network.log)
[performance.log](https://github.com/ppy/osu/files/8494286/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/8494287/performance-audio.log)

**Comment by Susko3** [comment]
If you change the monitor resolution in windows settings, do you experience the same thing?

**Reply by Rian8337** [reply]
The behavior is different.

If I change the monitor resolution while the game is running, the game ignores the fact that the monitor resolution has changed (this isn't the case when I tested with a maximized and fullscreen'd Microsoft Edge window). So if for example my game is running at `1920x1080` resolution, then I changed my monitor resolution to `1680x1050` in Windows settings, the game will stay at `1920x1080` until I restart it.

When testing that, I set the in-game resolution setting to `Default`. With other resolution options, the game stays at the resolution I picked, which is aligned with my expectation.
